instfixlib.sh

```diff
@@ -0,0 +1,535 @@
+	if [ $? -ne 0 ]; then
+		# User does not exist
+		$SUDOX useradd -m -s /usr/sbin/nologin "$username"
+		echo "User $username created"
+	fi
+	# Add the current non-root user to the iobroker group so he can access the iobroker dir
+	if [ "$username" != "$USER" ] && [ "$IS_ROOT" = false ]; then
+		sudo usermod -a -G $username $USER
+	fi
+
+	SUDOERS_CONTENT="$username ALL=(ALL) ALL\n"
+	# Add the user to all groups we need and give him passwordless sudo privileges
+	# Define which commands iobroker may execute as sudo without password
+	declare -a iob_commands=(
+		"shutdown -h now" "halt" "poweroff" "reboot"
+		"systemctl start" "systemctl stop"
+		"mount" "umount" "systemd-run"
+		"apt-get" "apt" "dpkg" "make"
+		"ping" "fping"
+		"arp-scan"
+		"setcap"
+		"vcgencmd"
+		"cat"
+		"df"
+		"mysqldump"
+		"ldconfig"
+	)
+	add2sudoers "$username ALL=(ALL) " "${iob_commands[@]}"
+
+	# Additionally, define which iobroker-related commands may be executed by every user
+	declare -a all_user_commands=(
+		"systemctl start iobroker"
+		"systemctl stop iobroker"
+		"systemctl restart iobroker"
+	)
+	add2sudoers "ALL ALL=" "${all_user_commands[@]}"
+
+	# Furthermore, allow all users to execute node iobroker.js as iobroker
+	if [ "$IOB_USER" != "$USER" ]; then
+		add2sudoers "ALL ALL=($IOB_USER) " "node $CONTROLLER_DIR/iobroker.js"
+	fi
+
+	SUDOERS_FILE="/etc/sudoers.d/iobroker"
+
+#ADOE: IF (INSTALLER) THEN ...
+	$SUDOX rm -f $SUDOERS_FILE
+#ADOE: ENDIF
+
+	echo -e "$SUDOERS_CONTENT" > ~/temp_sudo_file
+	$SUDOX visudo -c -q -f ~/temp_sudo_file && \
+		$SUDOX chown root:$ROOT_GROUP ~/temp_sudo_file &&
+		$SUDOX chmod 440 ~/temp_sudo_file &&
+		$SUDOX mv ~/temp_sudo_file $SUDOERS_FILE &&
+		echo "Created $SUDOERS_FILE"
+	# Add the user to all groups if they exist
+	declare -a groups=(
+		audio
+		bluetooth
+		dialout
+		gpio
+		i2c
+		redis
+		tty
+	)
+	for grp in "${groups[@]}"; do
+		getent group $grp &> /dev/null && $SUDOX usermod -a -G $grp $username
+	done
+}
+
+create_user_freebsd() {
+	username="$1"
+	id "$username" &> /dev/null
+	if [ $? -ne 0 ]; then
+		# User does not exist
+		$SUDOX pw useradd -m -s /usr/sbin/nologin -n "$username"
+	fi
+	# Add the user to all groups we need and give him passwordless sudo privileges
+	# Define which commands may be executed as sudo without password
+	# TODO: Find out the correct paths on FreeBSD
+	# SUDOERS_FILE="/usr/local/etc/sudoers.d/iobroker"
+
+	# Add the user to all groups if they exist
+	declare -a groups=(
+		audio
+		bluetooth
+		dialout
+		gpio
+		i2c
+		redis
+		tty
+	)
+	for grp in "${groups[@]}"; do
+		getent group $grp && $SUDOX pw group mod $grp -m $username
+	done
+}
+
+fix_dir_permissions() {
+	# Give the user access to all necessary directories
+	# When autostart is enabled, we need to fix the permissions so that `iobroker` can access it
+	echo "Fixing directory permissions..."
+
+#ADOE: IF (INSTALLER) THEN ...
+	$SUDOX chown -R $IOB_USER:$IOB_USER $IOB_DIR
+#ADOE: ELSE IF (FIXER) THEN ...
```

Comment:
dito

Reply:
Dito, as commented above, for lines 196 to 205.
The INSTALLER and the FIXER differ in this points. And i am not able to decide, which one is correct or if we need none or both.

Reply:
The chown line should be there in both cases and use the `change_owner` method.
We can try changing the cache dir owner in the installer too.